lib/pact/consumer_contract/pact_file.rb

```diff
@@ -79,12 +79,37 @@ def get_remote_with_retry(uri_string, options)
 
     def get_remote(uri, options)
       request = Net::HTTP::Get.new(uri)
+      request = prepare_auth(request, options) if options[:username] || options[:token]
+
+      http = prepare_request(uri)
+      response = perform_http_request(http, request, options)
+
+      if response.is_a?(Net::HTTPRedirection)
+        uri = URI(response.header['location'])
+        req = Net::HTTP::Get.new(uri)
```

Comment:
I debated whether or not to make this a recursive call, but decided against it as it would introduce more complexity. But I'm open to adding it if you think more use-cases will be covered.